### PR TITLE
Transactions interpretation utils

### PR DESCRIPTION
**Pull request by razvantomegea**

### CODE REVIEW

**Review by CiprianDraghici — APPROVED** [review]

File: src/utils/transactions/parsers/parseTransactionAfterSigning.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Transaction, TransactionPayload } from '@multiversx/sdk-core/out';
+import { PlainSignedTransaction } from '@multiversx/sdk-web-wallet-provider/out/plainSignedTransaction';
+import { SignedTransactionType } from 'types';
+import { TransactionServerStatusesEnum } from 'types/enums.types';
+import { isGuardianTx } from '../validation';
+
+export function parseTransactionAfterSigning(
```
Comment (arhtudormorar) [inline src/utils/transactions/parsers/parseTransactionAfterSigning.ts:7]: is it needed?

File: src/utils/transactions/parsers/transactionActionUnwrapper/transactionActionUnwrapper.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { esdtNftUnwrapper, mexUnwrapper, stakeUnwrapper } from './helpers';
+import {
+  TransactionActionCategoryEnum,
+  TransactionActionType,
+  UnwrapperType
+} from '../../../../types';
```
Comment (arhtudormorar) [inline src/utils/transactions/parsers/transactionActionUnwrapper/transactionActionUnwrapper.ts:6]: ?

File: src/utils/walletconnect/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './__sdkWalletconnectProvider';
```
Comment (arhtudormorar) [inline src/utils/walletconnect/index.ts:1]: is needed?

File: src/utils/window/addOriginToLocationPath.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { getWindowLocation } from './getWindowLocation';
 
-export const addOriginToLocationPath = (path = '') => {
+export function addOriginToLocationPath(path = '') {
```
Comment (arhtudormorar) [inline src/utils/window/addOriginToLocationPath.ts:3]: why?

File: src/utils/window/clearNavigationHistory.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { getWindowLocation } from './getWindowLocation';
+
+export function clearNavigationHistory(remainingParams: any) {
```
Comment (arhtudormorar) [inline src/utils/window/clearNavigationHistory.ts:3]: why?

File: src/utils/window/getDefaultCallbackUrl.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { getWindowLocation } from './index';
 
-export const getDefaultCallbackUrl = () => {
+export function getDefaultCallbackUrl() {
```
Comment (arhtudormorar) [inline src/utils/window/getDefaultCallbackUrl.ts:3]: why?

File: src/utils/window/getIsAuthRoute.ts [hunk 7]
```diff
@@ -11,15 +11,12 @@ import { matchPath } from './matchPath';
     }
 ]
  */
-export const getIsAuthRoute = <
+export function getIsAuthRoute<
```
Comment (arhtudormorar) [inline src/utils/window/getIsAuthRoute.ts:14]: why?

File: src/utils/window/getWindowLocation.ts [hunk 8]
```diff
@@ -8,7 +8,7 @@ type GetWindowLocationType = {
   search: string;
 };
 
-export const getWindowLocation = (): GetWindowLocationType => {
+export function getWindowLocation(): GetWindowLocationType {
```
Comment (arhtudormorar) [inline src/utils/window/getWindowLocation.ts:11]: why?

File: src/utils/window/isWindowAvailable copy.ts [hunk 9]
```diff
@@ -1,2 +1,3 @@
-export const isWindowAvailable = () =>
-  typeof window != 'undefined' && typeof window?.location != 'undefined';
+export function isWindowAvailable() {
```
Comment (arhtudormorar) [inline src/utils/window/isWindowAvailable copy.ts:1]: Check the file name

File: src/utils/window/isWindowAvailable.ts [hunk 10]
```diff
@@ -1,2 +1,3 @@
-export const isWindowAvailable = () =>
-  typeof window != 'undefined' && typeof window?.location != 'undefined';
+export function isWindowAvailable() {
```
Comment (arhtudormorar) [inline src/utils/window/isWindowAvailable.ts:1]: why?

File: src/utils/window/parseNavigationParams.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  search: isWindowAvailable() ? window.location.search : '',
+  removeParams: []
+};
+
+/**
+ * @param preserveParams allows extracting params from URL search object
+ * @param options.removeParams allows removing params from URL search object
+ * @returns the selected params, search object with removed params, and the `clearNavigationHistory` helper
+ */
+export function parseNavigationParams(
```
Comment (arhtudormorar) [inline src/utils/window/parseNavigationParams.ts:19]: why?

File: src/utils/window/removeSearchParamsFromUrl.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import { clearNavigationHistory } from './clearNavigationHistory';
+import { isWindowAvailable } from './isWindowAvailable';
+import { parseNavigationParams } from './parseNavigationParams';
+
+interface RemoveSearchParamsFromUrlParamsType {
+  removeParams: string[];
+  search?: string;
+}
+
+export function removeSearchParamsFromUrl({
```
Comment (arhtudormorar) [inline src/utils/window/removeSearchParamsFromUrl.ts:11]: why?

File: src/utils/window/sanitizeCallbackUrl.ts [hunk 13]
```diff
@@ -1,7 +1,7 @@
-export const sanitizeCallbackUrl = (
+export function sanitizeCallbackUrl(
```
Comment (arhtudormorar) [inline src/utils/window/sanitizeCallbackUrl.ts:1]: why?

**Pull request closed without merging**